docs/proposals/karmadactl/enhance_karmadactl_maintenance_experience.md

```diff
@@ -0,0 +1,306 @@
+
+- performance：
+<ol>
+<li>When the `clusters` flag is empty and the `all-clusters` flag is set to false, a message should prompt the user with "Please specify at least one target cluster," as would occur with a command like `karmadactl top pod`.</li>
+<li>When the `all-clusters` flag is set to true, the command's scope covers all member clusters. For example, running `karmadactl top pod -C member1 --all-clusters` would show metrics for pods across all member clusters.</li>
+<li>When the `clusters` flag is not empty, the command's scope is determined by the member clusters specified in the `clusters` flag. For instance, the command `karmadactl top pod -C member1,member2` would show metrics for pods from the member clusters member1 and member2.</li>
+</ol>
+
+### Adding New Commands
+
+#### karmadactl edit, create, delete, label, patch and annotate
+
+These commands provide the capability for karmadactl to create, delete, and update resources on the Karmada control plane with functionalities that are consistent with those available in kubectl.
+
+- usage scope: Karmada control plane
+- scope-related flag set: None.
+
+#### karmadactl top node
+
+- description: Display resource (CPU/memory) usage of nodes in member clusters.
+- usage scope: Multiple Member Clusters
+- scope-related flag set:
+
+| name         | shorthand | default | usage                                                  |
+|--------------|-----------|---------|--------------------------------------------------------|
+| clusters     | C         | none    | Used to specify target member clusters                 |
+| all-clusters | /         | false   | Used to specify whether to include all member clusters |
+
+- performance:
+<ol>
+<li>When the `clusters` flag is empty and the `all-clusters` flag is set to false, a message should prompt the user with "Please specify at least one target cluster," as would occur with a command like `karmadactl top node`.</li>
+<li>When the `all-clusters` flag is set to true, the command's scope covers all member clusters. For example, running `karmadactl top node -C member1 --all-clusters` would show metrics for nodes across all member clusters.</li>
+<li>When the `clusters` flag is not empty, the command's scope is determined by the member clusters specified in the `clusters` flag. For instance, the command `karmadactl top node -C member1,member2` would show metrics for nodes from the member clusters member1 and member2.</li>
+</ol>
+
+#### karmadactl attach
+- description: attach to a process that is already running inside an existing container in a member cluster.
+- usage scope: A Single Member Cluster
+- scope-related flag set:
+
+| name        | shorthand    | default | usage                                                |
+|-------------|--------------|---------|------------------------------------------------------|
+| cluster     | C            | none    | Used to specify target member cluster                |
```

Comment:
It seems the `-C` parameter represents both the `--cluster` and -`-clusters` options.

Reply:
Yes, since options `clusters` and `cluster` do not appear at the same time, I choose to adopte the existing implementation and use the shorthand `-C` to represent both of them. Do you have any suggestions?

Reply:
The `-C` option is not introduced by this requirement; it's part of the existing implementation. I think the new command's option `cluster/clusters` can be set without a shorthand, and we can leave the existing commands as they are for now.